src/jit/gentree.cpp

```diff
@@ -18151,6 +18153,20 @@ GenTree* Compiler::gtNewMustThrowException(unsigned helper, var_types type, CORI
     }
     return node;
 }
+
+// Returns true for the xarch MemoryStore SIMD HW Instrinsic instructions, false otherwise
+bool GenTreeHWIntrinsic::OperIsMemoryStore()
+{
+#ifdef _TARGET_XARCH_
+    HWIntrinsicCategory category = Compiler::categoryOfHWIntrinsic(gtHWIntrinsicId);
+    if (category == HW_Category_MemoryStore)
```

Comment:
This check is not enough because of `Avx/Avx2.ExtractVector128` which have the category of `HW_Catory_IMM`. I suggest changing here to 
```cpp
if (category == HW_Category_MemoryStore || ((gtHWIntrinsicId == AVX_ExtractVector128 || gtHWIntrinsicId == AVX2_ExtractVector128) && Compiler::numArgsOfHWIntrinsic(this) == 3))
```

Reply:
OK I will add that

Reply:
Yes, but we have to set the category as `HW_Category_IMM` now...

Reply:
I think I would prefer to add a new flag:

'HW_Flag_WriteMemory,'

Reply:
I am okay to add this flag, but it does not help here. Because `Avx/Avx2.ExtractVector128` have `MemoryStore` and non-`MemoryStore` overloads both, `((gtHWIntrinsicId == AVX_ExtractVector128 || gtHWIntrinsicId == AVX2_ExtractVector128) && Compiler::numArgsOfHWIntrinsic(this) == 3)` cannot be aovided.

Reply:
For example:
```csharp
public static Vector128<sbyte> ExtractVector128(Vector256<sbyte> value, byte index);
public static unsafe void ExtractVector128(sbyte* address, Vector256<sbyte> value, byte index);
```

Reply:
A new flag would be helpful if we check the first op's type.
```cpp
if ((flags & HW_Flag_WriteMemory) != 0 && op1->TypeGet() == TYP_I_IMPL)
```